src/test/messageHandlers/JSONHandlerTest.java

```diff
@@ -0,0 +1,39 @@
+package messageHandlers;
+
+import clientHandlers.JSONExtractor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JSONHandlerTest{
+    JSONExtractor jsonEx;
+    String testMessageOpen;
+    String testMessageClose;
+
+    @BeforeEach
+    void setUp() {
+        jsonEx = new JSONExtractor();
+        testMessageOpen = "{\"action\": {\"id\": 1, \"type\": \"openLock\"}}";
+        // testMessage is the example message on the message board in Basecamp.
+        testMessageClose = "{\"action\": {\"id\": 2, \"type\": \"closeLock\"}}";
+    }
+
+    @Test
+    void testMessageHandlerLock() {
+        assertEquals("openLock",jsonEx.handleMessage(testMessageOpen).getAction());
```

Comment:
Separate into different test. Otherwise we do not know in which case they go wrong